spock-specs/src/test/groovy/org/spockframework/mock/runtime/ByteBuddyMockFactoryConcurrentSpec.groovy

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.mock.runtime
+
+import groovy.transform.Canonical
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Timeout
+import spock.lang.Unroll
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Phaser
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+
+class ByteBuddyMockFactoryConcurrentSpec extends Specification {
+  private static final String IfA = "IfA"
+  private static final String IfB = "IfB"
+  private static final String IfC = "IfC"
+  @Shared
+  final IProxyBasedMockInterceptor interceptor = Stub()
+
+  // Just to be save to abort, normally the tests run in 2 secs.
+  @Timeout(40)
+  @Unroll("#test")
+  def "cacheLockingStressTest"(String test, MockSpec mockSpecA, MockSpec mockSpecB) {
```

Comment:
iirc, we usually do not declare the data variables explicitly except when testing that functionality.
And especially not ones only used in the naming pattern like `test`.